src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/configs/CacheActionConfig.java

```diff
@@ -68,7 +68,7 @@ public CacheActionConfig(RcaConf conf) {
     cacheSettingsConfig = new NestedConfig("cache-settings", actionConfig);
     fieldDataCacheConfig = new FieldDataCacheConfig(cacheSettingsConfig);
     shardRequestCacheConfig = new ShardRequestCacheConfig(cacheSettingsConfig);
-    totalStepCount = new Config<>(TOTAL_STEP_COUNT_CONFIG_NAME, cacheSettingsConfig.getValue(),
+    totalStepCount = new Config<>(TOTAL_STEP_COUNT_CONFIG_NAME, actionConfig,
```

Comment:
step count will be a part of cacheSettingConfig right? Is there a reason why we are moving to actionConfig?
Will it be common parameter for cache and queue?

Reply:
step count is indeed part of the cacheSettingConfig. But here we need to pass the parent Map<> object as a parameter in the constructor of the Config class